httpstats/context.go

```diff
@@ -0,0 +1,62 @@
+package httpstats
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/segmentio/stats"
+)
+
+// RequestWithTags returns a shallow copy of req with its context changed with this provided tags
+// so the they can be used later during the RoundTrip in the metrics recording.
+// The provided ctx must be non-nil.
+func RequestWithTags(req *http.Request, tags ...stats.Tag) *http.Request {
```

Comment:
Is there a use case where you'd want just the behavior of `RequestWithTags`, but not `AddTagsToRequest`? I'm wondering if we should just make `RequestWithTags` always append new tags, regardless of what tags (if any) are present.

Reply:
The problem is that `RequestWithTags` creates a shallow clone, necessary for passing that request into child handlers or transports.  `AddTagsToRequest` explicitly doesn't do that.

I think I would have created the API slightly differently (`RequestWithTags` just an initialization function) to make it less confusing, but I think we should maintain the backwards compatibility.

Reply:
After promoting context to the stats package at @achille-roussel 's suggestion, this comes much more naturally.  :raised_hands: